ownCloudSDK/Connection/OCConnection.m

```diff
@@ -1405,9 +1414,9 @@ - (NSProgress *)connectWithCompletionHandler:(void(^)(NSError *error, OCIssue *i
 				}
 			};
 		};
-
+    
 		// Check status
-		if ((statusRequest =  [OCHTTPRequest requestWithURL:[self URLForEndpoint:OCConnectionEndpointIDStatus options:nil]]) != nil)
+		if ((statusRequest = [OCHTTPRequest requestWithURL:[self URLForEndpoint:OCConnectionEndpointIDStatus options:nil]]) != nil || (statusRequest = [OCHTTPRequest requestWithURL:[self URLForEndpoint:OCConnectionEndpointIDKiteworksStatus options:nil]]) != nil)
```

Comment:
As described in another comment already, this construct looks risky and unreliable.

Since the OCBookmark, at this point, should already be able to tell if this is a Kiteworks server and, at `ownCloudSDK/Connection/OCConnection+Tools.m:70`, you're prepending `kwdav` to all but the well-known URL in that case, `OCConnectionEndpointIDStatus` should already return `kwdav/status.php`, so `OCConnectionEndpointIDKiteworksStatus` shouldn't be needed here.

If it wasn't so, my next suggestion would have been to determine which URL to check depending on `OCConnection.isKiteworksServer` - and then only check that.